buildSrc/src/main/kotlin/JsPlugin.kt

```diff
@@ -0,0 +1,104 @@
+import com.moowork.gradle.node.NodeExtension
+import com.moowork.gradle.node.NodePlugin
+import com.moowork.gradle.node.npm.NpmTask
+import com.moowork.gradle.node.task.NodeTask
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.repositories.IvyArtifactRepository
+import org.gradle.api.internal.AbstractTask
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.jetbrains.kotlin.gradle.dsl.KotlinJsOptions
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.tasks.Kotlin2JsCompile
+import java.io.File
+
+@Suppress("UnstableApiUsage")
+abstract class JsPlugin : Plugin<Project> {
+
+    override fun apply(target: Project) {
+        configureJsCompilation(target)
+        configureJsTest(target)
+    }
+
+    private fun configureJsCompilation(target: Project) {
+        target.extensions.configure(KotlinMultiplatformExtension::class.java) {
+            targetFromPreset(presets.getByName("js"), "js")
+            sourceSets.getByName("jsMain").dependencies {
+                implementation(kotlin("stdlib-js"))
+            }
+            sourceSets.getByName("jsTest").dependencies {
+                implementation(kotlin("test-js"))
+            }
+        }
+        target.tasks.named("compileKotlinJs", Kotlin2JsCompile::class.java) {
+            kotlinOptions.configure()
+        }
+        target.tasks.named("compileTestKotlinJs", Kotlin2JsCompile::class.java) {
+            kotlinOptions.configure()
+        }
+    }
+
+    private fun configureJsTest(target: Project) {
+        // workaround for https://github.com/srs/gradle-node-plugin/issues/301
+        target.repositories.whenObjectAdded {
+            if (this is IvyArtifactRepository) {
+                metadataSources {
+                    artifact()
+                }
+            }
+        }
+
+        target.pluginManager.apply(NodePlugin::class.java)
+        target.extensions.configure(NodeExtension::class.java) {
+            download = true
+            version = "12.2.0"
+        }
+
+        val dependenciesTaskProvider = target.tasks.register("installJsTestDependencies", NpmTask::class.java) {
+            setArgs(listOf("install", "kotlin@1.3.31", "kotlin-test@1.3.31", "mocha@6.1.4"))
```

Comment:
let's use `ext.kotlin_version` variable?

Reply:
Extracted to `Versions.kotlin`, ext `build.gradle` is not available in `buildSrc`.